files/docker/node/addons/entrypoint.sh

```diff
@@ -44,26 +44,28 @@ return 0
 
 export UPDATE_CHECK='N'
 
+[[ -z ${G_BRANCH} ]] && G_BRANCH="master"
+
 if [[ "$NETWORK" == "mainnet" ]]; then
-  $CNODE_HOME/scripts/guild-deploy.sh -n mainnet -u -s f > /dev/null 2>&1 \
+  $CNODE_HOME/scripts/guild-deploy.sh -b ${G_BRANCH} -n mainnet -u -s f > /dev/null 2>&1 \
   && customise \
   && exec $CNODE_HOME/scripts/cnode.sh
 elif [[ "$NETWORK" == "preprod" ]]; then
-  $CNODE_HOME/scripts/guild-deploy.sh -n preprod -u -s f > /dev/null 2>&1 \
+  $CNODE_HOME/scripts/guild-deploy.sh -b ${G_BRANCH} -n preprod -u -s f > /dev/null 2>&1 \
   && customise \
   && exec $CNODE_HOME/scripts/cnode.sh
 elif [[ "$NETWORK" == "preview" ]]; then
-  $CNODE_HOME/scripts/guild-deploy.sh -n preview -u -s f > /dev/null 2>&1 \
+  $CNODE_HOME/scripts/guild-deploy.sh -b ${G_BRANCH} -n preview -u -s f > /dev/null 2>&1 \
   && customise \
   && exec $CNODE_HOME/scripts/cnode.sh
 elif [[ "$NETWORK" == "guild-mainnet" ]]; then
-  $CNODE_HOME/scripts/guild-deploy.sh -n mainnet -u -s f > /dev/null 2>&1 \
+  $CNODE_HOME/scripts/guild-deploy.sh -b ${G_BRANCH} -n mainnet -u -s f > /dev/null 2>&1 \
   && bash /home/guild/.scripts/guild-topology.sh > /dev/null 2>&1 \
   && export TOPOLOGY="${CNODE_HOME}/files/guildnet-topology.json" \
   && customise \
   && exec $CNODE_HOME/scripts/cnode.sh
 elif [[ "$NETWORK" == "guild" ]]; then
-  $CNODE_HOME/scripts/guild-deploy.sh -n guild -u -s f > /dev/null 2>&1 \
+  $CNODE_HOME/scripts/guild-deploy.sh -b ${G_BRANCH} -n guild -u -s f > /dev/null 2>&1 \
```

Comment:
building a fork from a container is broken because entrypoint.sh will always overwrite the images config files with the latest master. When building for 8.9.1 ( and any other release with breaking changes ing config files) the node fails to start.
re: https://github.com/Fuma419/guild-operators/blob/d2d9d9c777f4a3167da007e79d4355bcc957100c/files/docker/node/addons/entrypoint.sh#L50

Reply:
per Cloud Native best practices, static configuration file that reliably function are provided as default.

Reply:
> building a fork from a container is broken because entrypoint.sh will always overwrite the images config files with the latest master. When building for 8.9.1 ( and any other release with breaking changes in config files) the node fails to start. re: https://github.com/Fuma419/guild-operators/blob/d2d9d9c777f4a3167da007e79d4355bcc957100c/files/docker/node/addons/entrypoint.sh#L50
> 
> Per my above comment "This should be considered as untested and is not ready for production." please test and provide any specific concerns so this can be made production ready.

I wouldn't call this broken building from a fork as the container always does this. This is actually an artifact of the behavior I described above, and was the intended behavior implemented by other contributors. 

Have you tried following the instructions to mount files at `priv/files` as this is the only way to currently avoid the dynamically updated files causing the crash behavior? The entrypoint itself describes this when not provided a network: [Entrypoint suggesting a volume mounted to `$CNODE_HOME/priv/files`](https://github.com/cardano-community/guild-operators/pull/1747/files#diff-3669ebd3211a53429d9faed1d1da90c19076bd14380f86771193d9eaf3da5ce9R71-R76)

Reply:
> per Cloud Native best practices, static configuration files that reliably function are provided as default.

What I meant by static files was part of the logic of not dynamically replacing the built in files at all. The existing logic, and your changes, does not alter the container replacing them when the container starts. It simply provides more control over where the dynamic updates are coming from, but can still lead to replacing them which is behavior that is uncommon among container images.

Reply:
regarding this reference: [Entrypoint suggesting a volume mounted to $CNODE_HOME/priv/files](https://github.com/cardano-community/guild-operators/pull/1747/files#diff-3669ebd3211a53429d9faed1d1da90c19076bd14380f86771193d9eaf3da5ce9R71-R76)

There are a couple issues with these echo lines, and i will propose some corrections. 
1.  $CNODE_HOME/priv/files is not a valid path, but $CNODE_HOME/priv and $CNODE_HOME/files are.
2. mainnet-config.json is not the naming convention in cnode deployments. ref env: #CONFIG="${CNODE_HOME}/files/config.json"

Reply:
@Fuma419 Did you find the changes from #1754 acceptable for this purpose?

Reply:
Yes, the functionality described works for me. The tips are appreciated in docs/docker/tips.md. I will spend some time testing your this feature on alpha and using the .env override on my fork. 

I think the default value of UPDATE_CHECK=N is what i would personally prefer but it may or may not be ideal for the community as a whole. Would the purpose of UPDATE_CHECK=Y be to facilitate hotfixs to cfg files?

Reply:
I included `UPDATE_CHECK=N` in #1754 as the default for images via the Dockerfile's ENV variables. I feel this is the most "cloud native" way for the container to behave and quite simple for the user to change with multiple methods available.